### PR TITLE
updated to make metrics compatible with Solaris

### PR DESCRIPTION
**Pull request by makaveli0129**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

**Comment by majormoses** [comment]
I am not too familiar with `solaris` so thanks for putting this together. Just to be clear this does not include inode capacity metrics? Also could you please include a [changelog entry](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md) and a manual or automated [testing artifact](https://github.com/sensu-plugins/community/blob/master/PULL_REQUEST_PROCESS.md#7-testing-artifacts) for each of them to show me what the output looks like?

**Comment by makaveli0129** [comment]
# /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/metrics-disk-capacity.rb
corpt631.disk./devices.used 0 1524050659
corpt631.disk./devices.avail 0% 1524050659
corpt631.disk./devices.capacity /devices 1524050659
corpt631.disk.vx/dsk/corpt631_vg/policy.used 9050252 1524050659
corpt631.disk.vx/dsk/corpt631_vg/policy.avail 47% 1524050659
corpt631.disk.vx/dsk/corpt631_vg/policy.capacity /policy 1524050659
corpt631.disk.vx/dsk/corpt631_vg/ldap2.used 23070941 1524050659
corpt631.disk.vx/dsk/corpt631_vg/ldap2.avail 68% 1524050659
corpt631.disk.vx/dsk/corpt631_vg/ldap2.capacity /ldap2 1524050659
corpt631.disk.vx/dsk/corpt631_vg/ldap.used 16473049 1524050659
corpt631.disk.vx/dsk/corpt631_vg/ldap.avail 76% 1524050659
corpt631.disk.vx/dsk/corpt631_vg/ldap.capacity /ldap 1524050659

**Comment by makaveli0129** [comment]
# /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/metrics-disk-capacity.rb -s tester
tester.disk./devices.used 0 1524050741
tester.disk./devices.avail 0% 1524050741
tester.disk./devices.capacity /devices 1524050741
tester.disk.vx/dsk/corpt631_vg/policy.used 9050252 1524050741
tester.disk.vx/dsk/corpt631_vg/policy.avail 47% 1524050741
tester.disk.vx/dsk/corpt631_vg/policy.capacity /policy 1524050741
tester.disk.vx/dsk/corpt631_vg/ldap2.used 23070941 1524050741
tester.disk.vx/dsk/corpt631_vg/ldap2.avail 68% 1524050741
tester.disk.vx/dsk/corpt631_vg/ldap2.capacity /ldap2 1524050741
tester.disk.vx/dsk/corpt631_vg/ldap.used 16473049 1524050741
tester.disk.vx/dsk/corpt631_vg/ldap.avail 76% 1524050741
tester.disk.vx/dsk/corpt631_vg/ldap.capacity /ldap 1524050741

**Comment by makaveli0129** [comment]
# /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/metrics-disk-usage.rb -s tester
tester.root.used 30251565 1524050771
tester.root.avail 21554911 1524050771
tester.root.used_percentage 59 1524050771
tester.root.devices.used 0 1524050771
tester.root.devices.avail 0 1524050771
tester.root.devices.used_percentage 0 1524050771
tester.root.system.contract.used 0 1524050771
tester.root.system.contract.avail 0 1524050771
tester.root.system.contract.used_percentage 0 1524050771
tester.root.proc.used 0 1524050771
tester.root.proc.avail 0 1524050771
tester.root.proc.used_percentage 0 1524050771
tester.root.etc.mnttab.used 0 1524050771
tester.root.etc.mnttab.avail 0 1524050771
tester.root.etc.mnttab.used_percentage 0 1524050771
tester.root.etc.svc.volatile.used 528 1524050771
tester.root.etc.svc.volatile.avail 78182864 1524050771
tester.root.etc.svc.volatile.used_percentage 1 1524050771
tester.root.system.object.used 0 1524050771
tester.root.system.object.avail 0 1524050771
tester.root.system.object.used_percentage 0 1524050771
tester.root.etc.dfs.sharetab.used 0 1524050771
tester.root.etc.dfs.sharetab.avail 0 1524050771
tester.root.etc.dfs.sharetab.used_percentage 0 1524050771
tester.root.platform.sun4v.lib.libc_psr.so.1.used 30251565 1524050771
tester.root.platform.sun4v.lib.libc_psr.so.1.avail 21554911 1524050771
tester.root.platform.sun4v.lib.libc_psr.so.1.used_percentage 59 1524050771
tester.root.platform.sun4v.lib.sparcv9.libc_psr.so.1.used 30251565 1524050771
tester.root.platform.sun4v.lib.sparcv9.libc_psr.so.1.avail 21554911 1524050771
tester.root.platform.sun4v.lib.sparcv9.libc_psr.so.1.used_percentage 59 1524050771
tester.root.tmp.used 366032 1524050771
tester.root.tmp.avail 78182864 1524050771
tester.root.tmp.used_percentage 1 1524050771
tester.root.var.run.used 64 1524050771
tester.root.var.run.avail 78182864 1524050771
tester.root.var.run.used_percentage 1 1524050771
tester.root.policy.used 7990239 1524050771
tester.root.policy.avail 9050252 1524050771
tester.root.policy.used_percentage 47 1524050771
tester.root.ldap2.used 48638524 1524050771
tester.root.ldap2.avail 23070941 1524050771
tester.root.ldap2.used_percentage 68 1524050771
tester.root.ldap.used 52086239 1524050771
tester.root.ldap.avail 16473049 1524050771
tester.root.ldap.used_percentage 76 1524050771
tester.root.iam-nas.used 10565628 1524050771
tester.root.iam-nas.avail 2017284 1524050771
tester.root.iam-nas.used_percentage 84 1524050771
tester.root.archive.used 39235052 1524050771
tester.root.archive.avail 13193748 1524050771
tester.root.archive.used_percentage 75 1524050771
tester.root.rpool.used 97 1524050771
tester.root.rpool.avail 21554911 1524050771
tester.root.rpool.used_percentage 1 1524050771

**Comment by makaveli0129** [comment]
# /opt/sensu/embedded/bin/ruby /opt/sensu/embedded/bin/metrics-disk-capacity.rb -s tester
tester.disk./devices.used 0 1524050980
tester.disk./devices.avail 0% 1524050980
tester.disk./devices.capacity /devices 1524050980
tester.disk.vx/dsk/vg/policy.used 9050252 1524050980
tester.disk.vx/dsk/vg/policy.avail 47% 1524050980
tester.disk.vx/dsk/vg/policy.capacity /policy 1524050980
tester.disk.vx/dsk/vg/ldap2.used 23070941 1524050980
tester.disk.vx/dsk/vg/ldap2.avail 68% 1524050980
tester.disk.vx/dsk/vg/ldap2.capacity /ldap2 1524050980
tester.disk.vx/dsk/vg/ldap.used 16473049 1524050980
tester.disk.vx/dsk//ldap.avail 76% 1524050980
tester.disk.vx/dsk//ldap.capacity /ldap 1524050980

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ This CHANGELOG follows the format listed [here](https://github.com/sensu-plugins
 
 ## [Unreleased]
 
+- Updated to include Compatibility for solaris OS
```
[the remark this answers is not in the document]
Comment (makaveli0129) [reply CHANGELOG.md]: i updated this

File: bin/metrics-disk-capacity.rb [hunk 2]
```diff
@@ -83,28 +88,30 @@ def run
     end
 
     # Get inode capacity metrics
-    `df -Pi`.split("\n").drop(1).each do |line|
-      begin
-        fs, _inodes, used, avail, capacity, _mnt = line.split
+    if Gem::Platform.local.os != 'solaris'
```
[the remark this answers is not in the document]
Comment (makaveli0129) [reply bin/metrics-disk-capacity.rb]: so when i try to run the inode command (https://www.thegeekdiary.com/solaris-how-to-determine-number-of-freeused-inodes-in-a-file-system/) on a machine with solaris i get the below:

# df -oi
df: operation not applicable for FSType autofs
df: operation not applicable for FSType ctfs
df: operation not applicable for FSType devfs
df: operation not applicable for FSType fd
df: operation not applicable for FSType lofs
df: operation not applicable for FSType mntfs
df: operation not applicable for FSType nfs
df: operation not applicable for FSType objfs
df: operation not applicable for FSType proc
df: operation not applicable for FSType sharefs
df: operation not applicable for FSType tmpfs
UX:vxfs df: INFO: V-3-25518: Usage:
df [-F vxfs] [generic_options] [-o {s | snapstat | ckpt= | ckptall | vols}] [directory | special ...]
df: operation not applicable for FSType zfs

File: bin/metrics-disk-capacity.rb [hunk 3]
```diff
@@ -83,28 +88,30 @@ def run
...
-        if fs =~ /\/dev/
-          fs = fs.gsub('/dev/', '')
-          metrics = {
-            disk: {
-              "#{fs}.iused" => used,
-              "#{fs}.iavail" => avail,
-              "#{fs}.icapacity" => capacity.delete('%')
+          timestamp = Time.now.to_i
+          if fs =~ /\/dev/
+            fs = fs.gsub('/dev/', '')
```
[the remark this answers is not in the document]
Comment (makaveli0129) [reply bin/metrics-disk-capacity.rb]: made this change

File: bin/metrics-disk-usage.rb [hunk 4]
```diff
@@ -95,7 +95,13 @@ def run
     delim = config[:flatten] == true ? '_' : '.'
     # Get disk usage from df with used and avail in megabytes
     # #YELLOW
-    `df -PB#{config[:block_size]} #{config[:local] ? '-l' : ''}`.split("\n").drop(1).each do |line|
+    command = if Gem::Platform.local.os == 'solaris'
+                "df -k #{config[:local] ? '-l' : ''}"
```
[the remark this answers is not in the document]
Comment (makaveli0129) [reply bin/metrics-disk-usage.rb]: @majormoses not sure how to handle this one as it is explicitly defaulted to "M" so checking if it was defined will always return true. Unless we remove the "default" i don't know if there will be an explicit way to tell if the command ARG was passed? Ideas?

[the remark this answers is not in the document]
Comment (majormoses) [reply bin/metrics-disk-usage.rb]: Hmm you are right, I think there is a way but I am not sure how to do it quickly and omitting the option seems reasonable enough.

File: bin/metrics-disk-capacity.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (majormoses) [reply bin/metrics-disk-capacity.rb]: then we should call out in the changelog that we don't support checking inodes on solaris.

**Comment by makaveli0129** [comment]
@majormoses changes to changelog have been submitted

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-disk-checks/versions/3.1.0

**Pull request merged**